SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -580,6 +580,153 @@ void CFlowOutput::SetAnalyzeSurface(const CSolver *solver, const CGeometry *geom
+  }
+}
+
+void CFlowOutput::LoadHistoryData_Turb(const CConfig* config, const CSolver* const* solver) {
+  const auto turb_solver = solver[TURB_SOL];
+
+  switch(config->GetKind_Turb_Model()){
+    case TURB_MODEL::SA: case TURB_MODEL::SA_NEG: case TURB_MODEL::SA_E: case TURB_MODEL::SA_COMP: case TURB_MODEL::SA_E_COMP:
+      SetHistoryOutputValue("RMS_NU_TILDE", log10(turb_solver->GetRes_RMS(0)));
+      SetHistoryOutputValue("MAX_NU_TILDE", log10(turb_solver->GetRes_Max(0)));
+      if (multiZone) {
+        SetHistoryOutputValue("BGS_NU_TILDE", log10(turb_solver->GetRes_BGS(0)));
+      }
+      break;
+    case TURB_MODEL::SST: case TURB_MODEL::SST_SUST:
+      SetHistoryOutputValue("RMS_TKE", log10(turb_solver->GetRes_RMS(0)));
+      SetHistoryOutputValue("RMS_DISSIPATION",    log10(turb_solver->GetRes_RMS(1)));
+      SetHistoryOutputValue("MAX_TKE", log10(turb_solver->GetRes_Max(0)));
+      SetHistoryOutputValue("MAX_DISSIPATION",    log10(turb_solver->GetRes_Max(1)));
+      if (multiZone) {
+        SetHistoryOutputValue("BGS_TKE", log10(turb_solver->GetRes_BGS(0)));
+        SetHistoryOutputValue("BGS_DISSIPATION",    log10(turb_solver->GetRes_BGS(1)));
+      }
+      break;
+    case TURB_MODEL::NONE: break;
+  }
+
+  if (config->GetKind_Turb_Model() != TURB_MODEL::NONE) {
+    SetHistoryOutputValue("LINSOL_ITER_TURB", turb_solver->GetIterLinSolver());
+    SetHistoryOutputValue("LINSOL_RESIDUAL_TURB", log10(turb_solver->GetResLinSolver()));
+  }
+}
+
+void CFlowOutput::SetVolumeOutputFields_Turb(const CConfig* config) {
+  switch(config->GetKind_Turb_Model()){
+    case TURB_MODEL::SA: case TURB_MODEL::SA_NEG: case TURB_MODEL::SA_E: case TURB_MODEL::SA_COMP: case TURB_MODEL::SA_E_COMP:
+      AddVolumeOutput("NU_TILDE", "Nu_Tilde", "SOLUTION", "Spalart-Allmaras variable");
+      AddVolumeOutput("RES_NU_TILDE", "Residual_Nu_Tilde", "RESIDUAL", "Residual of the Spalart-Allmaras variable");
+      if (config->GetKind_SlopeLimit_Turb() != NO_LIMITER) {
+        AddVolumeOutput("LIMITER_NU_TILDE", "Limiter_Nu_Tilde", "LIMITER", "Limiter value of the Spalart-Allmaras variable");
+      }
+      break;
+    case TURB_MODEL::SST: case TURB_MODEL::SST_SUST:
+      AddVolumeOutput("TKE", "Turb_Kin_Energy", "SOLUTION", "Turbulent kinetic energy");
+      AddVolumeOutput("DISSIPATION", "Omega", "SOLUTION", "Rate of dissipation");
+      AddVolumeOutput("RES_TKE", "Residual_TKE", "RESIDUAL", "Residual of turbulent kinetic energy");
+      AddVolumeOutput("RES_DISSIPATION", "Residual_Omega", "RESIDUAL", "Residual of the rate of dissipation");
+      if (config->GetKind_SlopeLimit_Turb() != NO_LIMITER) {
+        AddVolumeOutput("LIMITER_TKE", "Limiter_TKE", "LIMITER", "Limiter value of turb. kinetic energy");
+        AddVolumeOutput("LIMITER_DISSIPATION", "Limiter_Omega", "LIMITER", "Limiter value of dissipation rate");
```

Comment:
This is great but we probably need a finer-grained division because of the ordering problem, otherwise the restart files are not going to work

Reply:
Limiters are not relevant for restarts, right 🥟

Reply:
They are not, but I admit I like the residuals close to the residuals and the limiters close to the limiters :grimacing:

Reply:
Intermittency for transition model is also the solution to a transport equation (right?) ... so that should adhere to a restart order as well ... but I would need to check whether that is even written.
Man do I have no clue of transition modelling 🐔

Reply:
Transition would use CTurbSolver::LoadRestart and would therefore read crap anyway atm .. maybe catch Trans-restarts in the cfg-setpost

Reply:
No, intermittency is a derived quantity

Reply:
c050e57ce3c384e366d590239fb2c1d697192f0d adds the distinction for the volume TurbSolution and the rest (res+limiters)

Reply:
oc I always knew that the SA-BC model uses intermittency and does not require (to my knowledge) any more than the SA's nu_tilde. And as a certified transition expert I hope we have the revised SA-BCM implemented and not the original SA-BC ... the NASA-dudes even added a warning in red color(!) on [TMR](https://turbmodels.larc.nasa.gov/sa-bc_1eqn.html)

For the Langter-Mentry we would need this gamma an Re_theta value for restart but we have `SU2_MPI::Error("The LM transition model is under maintenance.", CURRENT_FUNCTION);` anyway

So I think transition model restarts are not an issue for now

Reply:
> They are not, but I admit I like the residuals close to the residuals and the limiters close to the limiters 😬

But... all those deletd lines 😢 ... fine I'll do the family reunion (tomorrow)